utils/windowfunction/wf_sum_avg.cpp

```diff
@@ -52,36 +52,50 @@ using namespace joblist;
 
 namespace windowfunction
 {
-template<typename T_IN, typename T_OUT>
-void WF_sum_avg<T_IN, T_OUT>::checkSumLimit(long double val, long double sum)
-{
-}
 
 template<typename T_IN, typename T_OUT>
-inline void WF_sum_avg<T_IN, T_OUT>::checkSumLimit(int128_t val, int128_t sum)
-{
-    if (((sum >= 0) && ((fMax128 - sum) < val)) ||
-        ((sum <  0) && ((fMin128 - sum) > val)))
-    {
-        string errStr = "SUM(int128_t)";
-        errStr = IDBErrorInfo::instance()->errorMsg(ERR_WF_OVERFLOW, errStr);
-        cerr << errStr << endl;
-        throw IDBExcept(errStr, ERR_WF_OVERFLOW);
-    }
-}
+inline void WF_sum_avg<T_IN,T_OUT>::checkSumLimit(T_IN& val, T_OUT& sum)
```

Comment:
Shouldn't the function parameters here be const ref?

Reply:
It should if it doesn't call for an explicit casts.